CommonTools/PileupAlgos/src/PuppiAlgo.cc

```diff
@@ -140,22 +167,57 @@ void PuppiAlgo::computeMedRMS(const unsigned int &iAlgo,const double &iPVFrac) {
     // some ways to do corrections to fRMS and fMedian
     fRMS [iAlgo] *= fRMSScaleFactor[iAlgo];
 
-    fRMS[iAlgo]    *= fRMSEtaSF;
-    fMedian[iAlgo] *= fMedEtaSF;
-    
-    if(!fAdjust[iAlgo]) return;
-    //Adjust the p-value to correspond to the median
-    std::sort(fPupsPV.begin(),fPupsPV.end());
-    int lNPV = 0; 
-    for(unsigned int i0 = 0; i0 < fPupsPV.size(); i0++) if(fPupsPV[i0] <= lMed ) lNPV++;
-    double lAdjust = double(lNPV)/double(fPupsPV.size()+fNCount[iAlgo]);
-    if(lAdjust > 0) fMedian[iAlgo] -= sqrt(ROOT::Math::chisquared_quantile(lAdjust,1.)*fRMS[iAlgo]);
+    if(fAdjust[iAlgo]){ 
+        //Adjust the p-value to correspond to the median
+        std::sort(fPupsPV.begin(),fPupsPV.end());
+        int lNPV = 0; 
+        for(unsigned int i0 = 0; i0 < fPupsPV.size(); i0++) if(fPupsPV[i0] <= lMed ) lNPV++;
+        double lAdjust = double(lNPV)/double(lNPV+0.5*fNCount[iAlgo]);
+        if(lAdjust > 0) {
+          fMedian[iAlgo] -= sqrt(ROOT::Math::chisquared_quantile(lAdjust,1.)*fRMS[iAlgo]);
+          fRMS[iAlgo]    -= sqrt(ROOT::Math::chisquared_quantile(lAdjust,1.)*fRMS[iAlgo]);
+        }        
+    }
+
+    // fRMS_perEta[iAlgo]    *= cur_RMSEtaSF;
+    // fMedian_perEta[iAlgo] *= cur_MedEtaSF;
+
+    for (unsigned int j0 = 0; j0 < fEtaMin.size(); j0++){
+        fRMS_perEta[iAlgo][j0] = fRMS[iAlgo]*fRMSEtaSF[j0];
+        fMedian_perEta[iAlgo][j0] = fMedian[iAlgo]*fMedEtaSF[j0];
+    }    
 
 }
 ////////////////////////////////////////////////////////////////////////////////
 
 //This code is probably a bit confusing
 double PuppiAlgo::compute(std::vector<double> const &iVals,double iChi2) const {
+    // if(fAlgoId[0] == -1) return 1;
```

Comment:
here and the rest of this PR, please cleanup the commented out code:
either you need it, then include it with some options so that it compiles and is available; or this is not really needed and should be removed.